rosidl_generator_cpp/test/test_array_generator.hpp

```diff
@@ -57,63 +57,25 @@ template<
   typename C,
   typename std::enable_if<
     std::is_integral<typename C::value_type>::value &&
-    !std::is_same<typename C::value_type, bool>::value &&
-    !std::is_same<typename C::value_type, char>::value &&
-    !std::is_same<typename C::value_type, int8_t>::value &&
-    !std::is_same<typename C::value_type, uint8_t>::value
+    !std::is_same<typename C::value_type, bool>::value
   >::type * = nullptr
 >
 void test_vector_fill(C * container, size_t size,
   typename C::value_type min, typename C::value_type max)
 {
-  std::default_random_engine rand_generator;
-  std::uniform_int_distribution<typename C::value_type> randnum(min, max);
-
-  if (size > 0) {
+  if (size > 0 && min != max) {
+    typename C::value_type step = (max - min) / size;
     (*container)[0] = min;
     for (size_t i = 1; i < size - 1; i++) {
-      (*container)[i] = randnum(rand_generator);
-    }
-    (*container)[size - 1] = max;
-  }
-}
-
-/**
- * Helper function to generate a test pattern for char type and derivatives.
- * Note: this is necessary because uniform_int_distribution is not defined for
- * char type.
- * Mininum and maximum values for the type and random values in the middle.
- * @param C Container (vector, array, etc) to be filled
- * @param size How many elements to fill in. Must size<=container_size
- * @param min Minimum value in the range to fill.
- * @param max Maximum value in the range to fill.
- */
-template<
-  typename C,
-  typename std::enable_if<
-    std::is_same<typename C::value_type, char>::value ||
-    std::is_same<typename C::value_type, int8_t>::value ||
-    std::is_same<typename C::value_type, uint8_t>::value
-  >::type * = nullptr
->
-void test_vector_fill(C * container, size_t size,
-  typename C::value_type min, typename C::value_type max)
-{
-  std::default_random_engine rand_generator;
-  std::uniform_int_distribution<int> randnum(static_cast<int>(min), static_cast<int>(max));
-
-  if (size > 0) {
-    (*container)[0] = min;
-    for (size_t i = 1; i < size - 1; i++) {
-      (*container)[i] = static_cast<typename C::value_type>(randnum(rand_generator));
+      (*container)[i] = min + i * step;
```

Comment:
Oh, I see. Too far away on this page :wink: Maybe remove the phrase "evenly distributed" from the docblock above then?